service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/authentication/StaticSasTokenProvider.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.service.digitaltwin.authentication;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+@AllArgsConstructor
+public class StaticSasTokenProvider implements SasTokenProvider {
+
+    @NonNull
+    private final String sharedAccessSignature;
```

Comment:
how does this ever get set?

Reply:
very good question

Reply:
return new StaticSasTokenProvider(sharedAccessSignature); 
in ServiceConnectionString.java

@AllArgsConstructor generates a constructor with 1 parameter for each field in your class. Fields marked with @NonNull result in null checks on those parameters.